includes/admin.form.inc

```diff
@@ -76,6 +76,61 @@ function islandora_large_image_admin(array $form, array &$form_state) {
       ),
     ),
   );
+  $form['islandora_large_image_thumbnail_fieldset'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Thumbnail'),
+    '#description' => t('Settings for creating thumbnail derivatives<br/><span style="color:red">WARNING: This will set the size for all medium sized images generated as a derivative. Regenerate derivatives to apply this setting to existing objects.</span>'),
+  );
+  $form['islandora_large_image_thumbnail_fieldset']['islandora_pdf_thumbnail_width'] = array(
+    '#type' => 'textfield',
+    '#attributes' => array(
+      'type' => 'number',
+    ),
+    '#element_validate' => array('element_validate_number'),
+    '#title' => t('Maximum Width'),
+    '#description' => t('The width of the thumbnail in pixels.'),
+    '#default_value' => variable_get('islandora_pdf_thumbnail_width', 200),
+    '#size' => 5,
+  );
+  $form['islandora_large_image_thumbnail_fieldset']['islandora_pdf_thumbnail_height'] = array(
+    '#type' => 'textfield',
+    '#attributes' => array(
+      'type' => 'number',
+    ),
+    '#element_validate' => array('element_validate_number'),
+    '#title' => t('Maximum Height'),
+    '#description' => t('The height of the thumbnail in pixels.'),
+    '#default_value' => variable_get('islandora_pdf_thumbnail_height', 200),
+    '#size' => 5,
+  );
+  $form['islandora_large_image_preview_fieldset'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Preview image'),
+    '#description' => t('Settings for creating preview image derivatives<br/><span style="color:red">WARNING: This will set the size for all medium sized images generated as a derivative. Regenerate derivatives to apply this setting to existing objects.</span>'),
```

Comment:
Should it define what a medium size image is and how during the creation of a medium size image derivative for PDFs share the size parameter for consistency? Sorry, I hope this isn't coming off wrong. I'm looking for a little "word-smithing" here, in hope to better word or identify if something really should change.

Reply:
I'm really not sure how best to handle the phrasing for this, since we're now essentially dynamically scoping these variables to whatever might use them... almost thinking there may be a better way to handle this: parameterize the dimensions in all the relevant derivative functions, and then use some [`hook_islandora_derivative_alter()`](https://github.com/Islandora/islandora/blob/e3bf90ea607aa19a5b769c5dd8caa4c923f1a242/islandora.api.php#L753-L773) implementation to inject our configuration from another module... the module which performs the alters can then describe the properties however it wants to, as it knows what derivative implementations it alters.

Reply:
Implementing the regeneration of all large image TN sounds potentially like a lot more code. Should that logic happen in a 2nd PR?